test/k8sT/Services.go

```diff
@@ -385,7 +385,7 @@ var _ = Describe("K8sServicesTest", func() {
 		failRequests := func(url string, count int, fromPod string) {
 			By("Making %d curl requests from %s to %q", count, fromPod, url)
 			for i := 1; i <= count; i++ {
-				res, err := kubectl.ExecInHostNetNS(context.TODO(), fromPod, helpers.CurlFail(url, "--max-time 3"))
+				res, err := kubectl.ExecInHostNetNS(context.TODO(), fromPod, helpers.CurlFail(url))
```

Comment:
Why is this changed? Can this be cause of our test fail?

Reply:
From the commit log:
```
helpers.CurlFail() can take optional arguments, and at one location this
is used to pass the "--max-time" option to the curl command line.

But these optional arguments for helpers.CurlFail() are not intended to
be used that way; they should instead be used to format the target URL,
which is supposed to be a string containing the relevant format
specifiers. Because in the current case, the URL provided has no format
specifier, go appends the additional arguments at the end, with
extraneous characters. So instead of producing the final curl command:

   curl [...] http://192.168.36.11:30861 --max-time 3 [...]

We actually get:

   curl [...] --max-time 8 http://192.168.36.11:30861%!(EXTRA string=--max-time 3)

The first "--max-time 8" is added by helpers.CurlFail() itself.

Then of course bash fails to interpret that command, and complains with:

   bash: -c: line 0: syntax error near unexpected token `('

In order to fix it, let's remove the "--max-time 3" argument entirely.
As we can see from the resulting command line, "--max-time" is already
used (with "CurlMaxTimeout") anyway. We will wait a few seconds more,
but at least the curl command will effectively be run.
```

So with that change, the `curl` command _is_ run (previously bash would fail to parse the command and would not run it), and this apparently caused the test to fail (`curl` passed when it was supposed to fail). But this only means that the test was not run correctly so far. So if the `curl` command runs and fail, I understand that either the test is incorrect, or it raises a bug in Cilium. I haven't found time to investigate further for now.

Reply:
Apologies for the late reply, I did not see the mention before somehow. I think this fix is okay for now to unblock people, but it would be great to have a configurable `CurlMaxTimeout`, since this change will add 30 seconds (6*5 additional seconds) to the test time.

Unfortunately, the only way `failRequest` will actually fail is due to a timeout, so it will wait 8 seconds each invocation.

Reply:
It will even add 6×8 seconds, since the 3-second long timeout was skipped so far (command not run). Yeah I can probably work out something to pass a configurable value. Could you please have a look at the other commit that changes some of the `failRequests()` to `doRequests()`?

Reply:
>  Could you please have a look at the other commit that changes some of the failRequests() to doRequests()?

I assume you are referring to https://github.com/cilium/cilium/pull/11307/commits/0be2205493bbbf0d261cc7a5abb1d86c9f1c9176 ?

If kube-proxy indeed forwards requests from local pods, then I think our implementation in NodePort BPF in the datapath is actually not compliant. But I haven't worked on that code in a while, @brb would likely know more.

Reply:
> I think this fix is okay for now to unblock people, but it would be great to have a configurable `CurlMaxTimeout`, since this change will add 30 seconds (6*5 additional seconds) to the test time.
> 
> Unfortunately, the only way `failRequest` should actually fail is due to a timeout, so it will wait 8 seconds each invocation.

I had another look and this is in fact not the case, because we should timeout on the value passed to `--connect-timeout` (5 seconds) here and not the one from `--max-time`. Test logs confirm that. So no change needed on that side.